src/ocean_data_assim/MOM_oda_driver.F90

```diff
@@ -514,44 +636,152 @@ subroutine set_analysis_time(Time,CS)
 end subroutine set_analysis_time
 
 !> Write observation differences to a file
-subroutine save_obs_diff(filename,CS)
-  character(len=*), intent(in) :: filename !< name of output file
-  type(ODA_CS), pointer, intent(in) :: CS  !< pointer to DA control structure
+! subroutine save_obs_diff(filename,CS)
+!   character(len=*), intent(in) :: filename !< name of output file
+!   type(ODA_CS), pointer, intent(in) :: CS  !< pointer to DA control structure
 
-  integer :: fid ! profile file handle
-  type(ocean_profile_type), pointer :: Prof=>NULL()
+!   integer :: fid ! profile file handle
+!   type(ocean_profile_type), pointer :: Prof=>NULL()
 
-  fid = open_profile_file(trim(filename), nvar=2, thread=SINGLE_FILE, fset=SINGLE_FILE)
-  Prof=>CS%CProfiles
+!   fid = open_profile_file(trim(filename), nvar=2, thread=SINGLE_FILE, fset=SINGLE_FILE)
+!   Prof=>CS%CProfiles
 
-  !! switch to global pelist
-  !call set_PElist(CS%filter_pelist)
+!   !! switch to global pelist
+!   !call set_PElist(CS%filter_pelist)
 
-  do while (associated(Prof))
-    call write_profile(fid,Prof)
-    Prof=>Prof%cnext
-  enddo
-  call close_profile_file(fid)
+!   do while (associated(Prof))
+!     call write_profile(fid,Prof)
+!     Prof=>Prof%cnext
+!   enddo
+!   call close_profile_file(fid)
 
-  !! switch back to ensemble member pelist
-  !call set_PElist(CS%ensemble_pelist(CS%ensemble_id,:))
+!   !! switch back to ensemble member pelist
+!   !call set_PElist(CS%ensemble_pelist(CS%ensemble_id,:))
 
-  return
-end subroutine save_obs_diff
+!   return
+! end subroutine save_obs_diff
```

Comment:
Did you want to remove this function?  If not, maybe a comment explaining why it is still here.